gadgets/trace_open/test/unit/trace_open_test.go

```diff
@@ -0,0 +1,271 @@
+	generateEvent func() (int, error)
+	validateEvent func(t *testing.T, info *utilstest.RunnerInfo, fd int, events []ExpectedTraceOpenEvent) error
+}
+
+func TestTraceOpenGadget(t *testing.T) {
+	utilstest.RequireRoot(t)
+	testCases := map[string]testDef{
+		"captures_all_events": {
+			runnerConfig: &utilstest.RunnerConfig{},
+			generateEvent: func() (int, error) {
+				return generateEvent(t)
+			},
+			validateEvent: func(t *testing.T, info *utilstest.RunnerInfo, fd int, events []ExpectedTraceOpenEvent) error {
+				utilstest.ExpectAtLeastOneEvent(func(info *utilstest.RunnerInfo, fd int) *ExpectedTraceOpenEvent {
+					return &ExpectedTraceOpenEvent{
+						Comm:  "unit.test",
+						Pid:   info.Pid,
+						Tid:   info.Tid,
+						Uid:   uint32(info.Uid),
+						Gid:   uint32(info.Gid),
+						Fd:    uint32(fd),
+						FName: "/dev/null",
+					}
+				})(t, info, fd, events)
+				return nil
+			},
+		},
+		"test_symbolic_links": {
+			runnerConfig: &utilstest.RunnerConfig{},
+			generateEvent: func() (int, error) {
+				// Create a symbolic link to /dev/null
+				err := os.Symlink("/dev/null", "/tmp/test_symbolic_links")
+				if err != nil {
+					return 0, fmt.Errorf("creating symbolic link: %w", err)
+				}
+				defer os.Remove("/tmp/test_symbolic_links")
+
+				// Open the symbolic link
+				fd, err := unix.Open("/tmp/test_symbolic_links", unix.O_RDONLY, 0)
+				if err != nil {
+					return 0, fmt.Errorf("opening file: %w", err)
+				}
+				defer unix.Close(fd)
+
+				return fd, nil
+			},
+			validateEvent: func(t *testing.T, info *utilstest.RunnerInfo, fd int, events []ExpectedTraceOpenEvent) error {
+				utilstest.ExpectAtLeastOneEvent(func(info *utilstest.RunnerInfo, fd int) *ExpectedTraceOpenEvent {
+					return &ExpectedTraceOpenEvent{
+						Comm:  "unit.test",
+						Pid:   info.Pid,
+						Tid:   info.Tid,
+						Uid:   uint32(info.Uid),
+						Gid:   uint32(info.Gid),
+						Fd:    uint32(fd),
+						FName: "/tmp/test_symbolic_links",
+					}
+				})(t, info, fd, events)
+				return nil
+			},
+		},
+		"test_relative_path": {
+			runnerConfig: &utilstest.RunnerConfig{},
+			generateEvent: func() (int, error) {
+				relPath := generateRelativePathForAbsolutePath(t, "/tmp/test_relative_path")
+				fd, err := unix.Open(relPath, unix.O_CREAT|unix.O_RDWR, unix.S_IRWXU|unix.S_IRGRP|unix.S_IWGRP|unix.S_IXOTH)
+				require.NoError(t, err, "opening file")
+
+				defer os.Remove(relPath)
+
+				unix.Close(fd)
+
+				return fd, nil
+			},
+			validateEvent: func(t *testing.T, info *utilstest.RunnerInfo, fd int, events []ExpectedTraceOpenEvent) error {
+				utilstest.ExpectAtLeastOneEvent(func(info *utilstest.RunnerInfo, fd int) *ExpectedTraceOpenEvent {
+					return &ExpectedTraceOpenEvent{
+						Comm:  "unit.test",
+						Pid:   info.Pid,
+						Tid:   info.Tid,
+						Uid:   uint32(info.Uid),
+						Gid:   uint32(info.Gid),
+						Fd:    uint32(fd),
+						FName: "../../../../../../../tmp/test_relative_path",
```

Comment:
this one fails for me. Should it use `generateRelativePathForAbsolutePath` instead of hardcoding the path?

Reply:
@mauriciovasquezbernal I am not sure it is best way or not the older build in were implemented this way, also again not sure the test seems to pass locally when i tested locally but I'll look into it.

Reply:
I think we need to use `generateRelativePathForAbsolutePath` otherwise it'll be depending on the location of the test on your machine, that's probably why it works for you and not for me.